design/ceph/object/ceph-cosi-driver.md

```diff
@@ -0,0 +1,161 @@
+# Ceph COSI Driver Support
+
+## Targeted for v1.12
+
+## Background
+
+Container Object Storage Interface (COSI) is a specification for container orchestration frameworks to manage object storage. Even though there is no standard protocol defined for Object Store, it has flexibility to add support for all. The COSI spec abstracts common storage features such as create/delete buckets, grant/revoke access to buckets, attach/detach buckets, and more. COSI released v1alpha1 with Kubernetes 1.25 and is not yet fully integrated by Kubernetes like CSI.
+More details about COSI can be found [here](https://kubernetes.io/blog/2022/09/02/cosi-kubernetes-object-storage-management/).
+It is projected that COSI will be the only supported object storage driver in the future and OBCs will be deprecated after some time.
+
+## COSI Driver Deployment
+
+The [COSI controller](https://github.com/kubernetes-sigs/container-object-storage-interface-controller) is deployed as container in the default namespace. It is not deployed by Rook.
+
+The Ceph COSI driver is deployed as a deployment with a [COSI sidecar container](https://github.com/kubernetes-sigs/container-object-storage-interface-provisioner-sidecar). The Ceph COSI driver is deployed in the same namespace as the Rook operator. The Ceph COSI driver is deployed with a service account that has the following RBAC permissions:
+
+## Integration plan with Rook
+
+The aim to support the v1alpha1 version of COSI in Rook v1.12. It will be extended to beta and release versions as appropriate. There will be one COSI Driver support from Rook. The driver will be started automatically with default settings when first Object Store gets created. The driver will be stopped when last Object Store is deleted. The driver will be deployed in the same namespace as Rook operator. The user can provide additional setting via new `CephCOSIDriver` CRD which is owned by Rook.
```

Comment:
> The driver will be stopped when last Object Store is deleted.

I would propose to follow the same pattern for uninstall as we do for the CSI driver. For CSI, an ownerReference is added to its resources that points to the Rook operator deployment. So the CSI driver is deleted whenever the Rook operator deployment is uninstalled. The same behavior seems best for the COSI driver, then Rook doesn't need any cleanup code for the COSI driver and its lifetime is the same as the operator.